detect_secrets/plugins/keyword.py

```diff
@@ -147,6 +176,17 @@
         secret=SECRET,
     ),
 )
+FOLLOWED_BY_EQUAL_SIGNS_OPTIONAL_BRACKETS_OPTIONAL_AT_SIGN_QUOTES_REQUIRED_REGEX = re.compile(
+    # e.g. my_password = "bar"
+    # e.g. my_password = @"bar"
+    # e.g. my_password[] = "bar";
```

Comment:
Based on below this is Objective-C. What does this actually do in Objective-C? (I couldn't Google it)

Reply:
@ means it is a constant
[] means it is a C style array e.g. `char myString[] = "This is a C character array";`

Something I do not handle yet, but can ticket, are strings like that following
```
password = [str1 stringByAppendingFormat:@"World"];
NSMutableString *password = [NSMutableString stringWithString:@"This string is mutable"];
```

Prior to this PR, we'd capture e.g. `[str1` and `[NSMutableString`, which was 🤐